plugins/USBPrinting/USBPrinterOutputDeviceManager.py

```diff
@@ -128,8 +128,19 @@ def getInstance(cls, engine = None, script_engine = None):
         return USBPrinterOutputDeviceManager._instance
 
     def _getDefaultFirmwareName(self):
-        machine_instance = Application.getInstance().getMachineManager().getActiveMachineInstance()
-        machine_type = machine_instance.getMachineDefinition().getId()
+        # Check whether getGlobalContainerStack() returns None or not...
+        global_container_stack = Application.getInstance().getGlobalContainerStack()
+        if not global_container_stack:
+            Logger.log("c", "getGlobalContainerStack() returned None")
+            Logger.log("i", "Closing firmware-updater UI as a consequence")
+            self._firmware_view.close()
+        
+        # Detecting id of the current machine
+        machine_id = global_container_stack.getBottom().id
```

Comment:
This comment seems a bit superfluous (we can read). If you want to add a comment here, say something like "the bottom of the containerstack is the machine definition")